### PR TITLE
Set aria-modal=true on Modal dialog so that Narrator can pick up on it in future releases

### LINKED ISSUE #6422: Modal: In scan mode, the narrator focus moves out of the Modal
#### Priorities and help requested:
Are you willing to submit a PR to fix? (No)
Requested priority: (Normal) 
 
#### Describe the issue:
 While navigating in scan mode, the narrator focus moves out of the modal. Tested on the sample modal component given in the Office UI Fabric website. (https://developer.microsoft.com/en-us/fabric#/components/modal)

Please check the images attached.

office ui fabric modal -1
office ui fabric modal - 2

## Todo

- [ ] Add best practices for including aria-hidden around non modal content when modal is open

### PR DESCRIPTION
**Pull request by aneeshack4**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #6422
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Right now, Narrator doesn't go inside the dialog of Modal components. When we contacted the Narrator team about this, they said it's because the dialog's not marked as modal and as a result, Narrator doesn't consider it a containment for navigation. They suggested using the aria-modal property to fix this but warned that though it should restrict screen reader focus to the dialog, in practice, it's not always implemented that way. Like their warning predicted, it didn't help Narrator announce the dialog or any of its contents. And it definitely didn't fix anything in Scan Mode. They said they're going to improve dialog reading in future versions. Another suggestion of theirs to trap narrator to the dialog was to add aria-hidden to the top level component because that'll extend magically to all the children. I experimented with this but didn't have much luck - you can see my implementation in this branch: https://github.com/aneeshack4/office-ui-fabric-react/tree/aria_hidden_modal
PR view to easily see which files changed: https://github.com/OfficeDev/office-ui-fabric-react/compare/master...aneeshack4:aria_hidden_modal?expand=1 

#### Focus areas to test

(optional)

### CODE REVIEW

**Comment by jspurlin** [comment]
Narrator should be reading out the dialog if it has a correct label on it... aria-modal shouldn't alter the way the content is initialy read out by defualt

**Comment by aneeshack4** [comment]
@jspurlin When I run narrator, it doesn't even go into the dialog. When we contacted the Narrator team about this, they suggested this aria-modal fix - their comments are at https://github.com/OfficeDev/office-ui-fabric-react/issues/6422#issuecomment-430331471

**Comment by aneeshack4** [comment]
@KevinTCoughlin hmm that does make sense thank you for providing the links to the documentation too! This example on w3 https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/dialog.html set aria-modal="true" even when role="dialog" was set so I thought I should do the same. But this didn't help solve the narrator problem either so I'm no longer sure this change's necessary... I guess I should go back to the Narrator team & ask them why the aria-modal solution was suggested when role="dialog" has already been set.

**Comment by jspurlin** [comment]
@aneeshack4 does the modal have a label? Also from the aria 1.1 spec on dialogs:

_Authors SHOULD provide a dialog label, which can be done with the aria-label or aria-labelledby attribute._

_Authors SHOULD ensure that all dialogs (both modal and non-modal) have at least one focusable descendant element. Authors SHOULD focus an element in the modal dialog when it is displayed, and authors SHOULD manage focus of modal dialogs._


The fact that `aria-modal` is set has nothing to do with if it will (or won't) be read out, it will at most only tell the screen reader if the dialog is modal (and potentially allow virtual focus move outside of the dialog). Although it sounds like screen readers may not automatically leave dialogs when navigating in scan mode (unless the user/app sets focus outside of the dialog).

@KevinTCoughlin while `aria-modal` is inherited into elements with `role="dialog"` it has the value of `false` by default so you need to pass `aria-modal="true"` if the dialog is in fact modal.

**Comment by aneeshack4** [comment]
@jspurlin Ahh I see... Inspect actually shows the ariaLabelledBy & ariaLabel attributes as not specified - they're set to {subtitleAriaId} and {titleAriaId} respectively in Modal.base.tsx. In Dialog.base.tsx, these two are generated programmatically inside corresponding methods this._getSubTextId() and this._getTitleTextId(). I'm not sure if I need to create separate methods like this or just assign to a string like 'titleariaid' - I searched for examples & the latter's being done but only in non base.tsx files (like test/stories/example). Any suggestions on which route I should go for Modal? 

And @KevinTCoughlin I misspoke - like Jeremy said, when I don't have aria-modal="true" explicitly, just having role="dialog" auto sets aria-modal to false.

**Comment by jspurlin** [comment]
The modal has `titleAriaId` and `subTitleAriaId` props If you pass those in and then render the content inside of the modal (applying the ids to the correct elements) you should be good to go

**Comment by dzearing** [comment]
This looks fine to me, but will defer to Kevin/Jeremy incase there are any leftover concerns. 

Ultimately we believe this is the correct attribute to put IRT the spec, we also see the screen readers not respecting it WRT scan mode. Narrator team has an open issue to make that work. But at least we're good from a spec standpoint.

File: packages/office-ui-fabric-react/src/components/Modal/examples/Modal.Basic.Example.tsx [hunk 1]
```diff
@@ -21,7 +21,14 @@ export class ModalBasicExample extends React.Component<
     return (
       <div>
         <DefaultButton secondaryText="Opens the Sample Modal" onClick={this._showModal} text="Open Modal" />
-        <Modal isOpen={this.state.showModal} onDismiss={this._closeModal} isBlocking={false} containerClassName="ms-modalExample-container">
+        <Modal
+          titleAriaId="Sample Modal"
+          subtitleAriaId="Modal Dialog example"
```
Comment (jspurlin) [inline packages/office-ui-fabric-react/src/components/Modal/examples/Modal.Basic.Example.tsx]: pretty sure this isn't what you intended... The `titleAriaId` and `subtitleAriaId` should be `id`s to elements, for example on the span that contains `'Lorem Ipsum'` on line 33 should have and `id`, like `title` and the `titleAriaId` would also be set to `title` (and a similar thing for the subtitle)

**Comment by aneeshack4** [comment]
just a note - still trying to figure out how to force Narrator's focus into the modal dialog (if it's even possible) because the fixing the aria labels wasn't enough.

**Comment by dzearing** [comment]
Following up on this one offline.

**Comment by sonimishra** [comment]
This issue is also linked with #6535 https://github.com/OfficeDev/office-ui-fabric-react/issues/6535 .

**Comment by sonimishra** [comment]
This issue is also linked with Power apps admin bug id https://github.com/OfficeDev/office-ui-fabric-react/issues/7480

**Comment by sonimishra** [comment]
Also linked with https://github.com/OfficeDev/office-ui-fabric-react/issues/7478

**Pull request merged**